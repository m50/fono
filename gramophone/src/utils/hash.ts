import { createHash } from 'crypto';

export function hash(algo: string, value: any) {
  const hashFunc = createHash(algo);
  return hashFunc.update(JSON.stringify(value)).digest('hex');
}

export function sha256(value: any) {
  return hash('sha256', value);
}

export function sha1(value: any) {
  return hash('sha1', value);
}

export function md5(value: any) {
  return hash('md5', value);
}
