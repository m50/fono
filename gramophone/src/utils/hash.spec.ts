import * as Hash from './hash';

describe('hash', () => {
  test('md5', () => {
    const result = Hash.md5('test');
    expect(result).toMatchSnapshot();
  });
  test('sha1', () => {
    const result = Hash.sha1('test');
    expect(result).toMatchSnapshot();
  });
  test('sha256', () => {
    const result = Hash.sha256('test');
    expect(result).toMatchSnapshot();
  });
});
