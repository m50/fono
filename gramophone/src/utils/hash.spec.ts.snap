// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hash md5 1`] = `"303b5c8988601647873b4ffd247d83cb"`;

exports[`hash sha1 1`] = `"5006d6f8302000e8b87fef5c50c071d6d97b4e88"`;

exports[`hash sha256 1`] = `"4d967a30111bf29f0eba01c448b375c1629b2fed01cdfcc3aed91f1b57d5dd5e"`;
